src/Databases/DatabaseOnDisk.cpp

```diff
@@ -369,6 +370,20 @@ void DatabaseOnDisk::checkMetadataFilenameAvailability(const String & to_table_n
 void DatabaseOnDisk::checkMetadataFilenameAvailabilityUnlocked(const String & to_table_name) const
 {
     String table_metadata_path = getObjectMetadataPath(to_table_name);
+    String suffix = ".sql.detached";
+    auto max_file_name_length = pathconf("/", _PC_NAME_MAX);
+
+    if (max_file_name_length == -1)
+        max_file_name_length = NAME_MAX;
+
+    //Max file name may be like 'database.table.sql.detached'
```

Comment:
This is not true. The detached flag name is `table.sql.detached` (without the database name). The longest filename is `/path/metadata_dropped/escaped_db_name.escaped_table_name.uuid.sql`: https://github.com/ClickHouse/ClickHouse/blob/426a51b624e03692ac418654c37b24e19e3f5d88/src/Interpreters/DatabaseCatalog.cpp#L1044-L1051